src/test/java/bio/terra/service/snapshot/SnapshotServiceTest.java

```diff
@@ -0,0 +1,179 @@
+            SnapshotRequestAccessInclude.SOURCES,
+            SnapshotRequestAccessInclude.TABLES,
+            SnapshotRequestAccessInclude.RELATIONSHIPS,
+            SnapshotRequestAccessInclude.PROFILE,
+            SnapshotRequestAccessInclude.DATA_PROJECT)),
+            equalTo(service.retrieveAvailableSnapshotModel(snapshotId)));
+    }
+
+    @Test
+    public void testRetrieveSnapshotOnlyAccessInfo() {
+        mockSnapshot();
+        assertThat(service.retrieveAvailableSnapshotModel(snapshotId, List.of(
+            SnapshotRequestAccessInclude.ACCESS_INFORMATION)),
+            equalTo(new SnapshotModel()
+                .id(snapshotId.toString())
+                .name(SNAPSHOT_NAME)
+                .description(SNAPSHOT_DESCRIPTION)
+                .createdDate(createdDate.toString())
+                .accessInformation(new SnapshotAccessInfoModel()
+                    .bigQuery(new SnapshotAccessInfoModelBigQuery()
+                        .datasetName(SNAPSHOT_NAME)
+                        .projectId(SNAPSHOT_DATA_PROJECT)
+                        .link("https://console.cloud.google.com/bigquery?project=" + SNAPSHOT_DATA_PROJECT)
+                        .tables(List.of(new SnapshotAccessInfoModelBigQueryTable()
+                            .name(SNAPSHOT_TABLE_NAME)
+                            .sampleQuery("SELECT * FROM `" + SNAPSHOT_DATA_PROJECT + "." + SNAPSHOT_NAME + "." +
+                                SNAPSHOT_TABLE_NAME + "` LIMIT 1000")
+                        ))
+                    )
+                )));
+    }
+
+    @Test
+    public void testRetrieveSnapshotMultiInfo() {
+        mockSnapshot();
+        assertThat(service.retrieveAvailableSnapshotModel(snapshotId, List.of(
+            SnapshotRequestAccessInclude.ACCESS_INFORMATION,
+            SnapshotRequestAccessInclude.SOURCES
+            )),
+            equalTo(new SnapshotModel()
+                .id(snapshotId.toString())
+                .name(SNAPSHOT_NAME)
+                .description(SNAPSHOT_DESCRIPTION)
+                .createdDate(createdDate.toString())
+                .source(Collections.emptyList())
+                .accessInformation(new SnapshotAccessInfoModel()
+                    .bigQuery(new SnapshotAccessInfoModelBigQuery()
+                        .datasetName(SNAPSHOT_NAME)
+                        .projectId(SNAPSHOT_DATA_PROJECT)
+                        .link("https://console.cloud.google.com/bigquery?project=" + SNAPSHOT_DATA_PROJECT)
+                        .tables(List.of(new SnapshotAccessInfoModelBigQueryTable()
+                            .name(SNAPSHOT_TABLE_NAME)
+                            .sampleQuery("SELECT * FROM `" + SNAPSHOT_DATA_PROJECT + "." + SNAPSHOT_NAME + "." +
+                                SNAPSHOT_TABLE_NAME + "` LIMIT 1000")
+                        ))
+                    )
+                )));
+    }
+
+    private void mockSnapshot() {
```

Comment:
In this case, the same mock data is used for all tests, so you don't need to reset and reinitialize the mocks between each test.

Reply:
It's not the same since I regenerate new timestamps and UUIDs.  I wanted to make sure to keep things separate.  Also, as we add more tests, this will likely not be the case

Reply:
> It's not the same since I regenerate new timestamps and UUIDs.

Also note that there's nothing about the UUID or timestamp values in this test that require them to be random or based on the current time. They could all be constant values and the test would still be valid.